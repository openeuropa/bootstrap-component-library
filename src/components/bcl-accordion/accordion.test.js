import { renderTwigFileAsNode } from "@openeuropa/bcl-test-utils";
import demoData from "@openeuropa/bcl-data-accordion/data";
import demoCustomTags from "@openeuropa/bcl-data-accordion/data--custom-tags";

const template = "@oe-bcl/bcl-accordion/accordion.html.twig";
const render = (params) => renderTwigFileAsNode(template, params);

describe("OE - Accordion", () => {
  test(`renders correctly`, () => {
    expect.assertions(1);

    return expect(render(demoData)).resolves.toMatchSnapshot();
  });

  test(`flush variant renders correctly`, () => {
    expect.assertions(1);

    return expect(
      render({ ...demoData, flush: true })
    ).resolves.toMatchSnapshot();
  });

  test(`with main title renders correctly`, () => {
    expect.assertions(1);

    return expect(
      render({
        ...demoData,
        title: "An accordion title",
        title_tag: "h4",
        title_link: {
          path: "/example.html",
          label: "A title created by the bcl heading template",
        },
      })
    ).resolves.toMatchSnapshot();
  });

  test(`with custom title tags renders correctly`, () => {
    expect.assertions(1);

    return expect(render(demoCustomTags)).resolves.toMatchSnapshot();
  });
});
