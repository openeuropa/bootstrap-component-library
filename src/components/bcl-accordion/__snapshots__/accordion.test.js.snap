// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OE - Accordion flush variant renders correctly 1`] = `
<jest>
  <div
    class="accordion accordion-flush"
    id="accordion-1"
  >
    <div
      class="accordion-item"
    >
      <h2
        class="accordion-header"
        id="heading-1-1"
      >
        <button
          aria-controls="collapse-1-1"
          aria-expanded="false"
          autocomplete="off"
          class="accordion-button"
          data-bs-target="#collapse-1-1"
          data-bs-toggle="collapse"
          type="button"
        >
          Accordion Item #1
        </button>
      </h2>
      <div
        aria-labelledby="heading-1-1"
        class="accordion-collapse collapse show"
        data-bs-parent="#accordion-1"
        id="collapse-1-1"
        role="region"
      >
        <div
          class="accordion-body"
        >
          Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.
        </div>
      </div>
    </div>
    <div
      class="accordion-item"
    >
      <h2
        class="accordion-header"
        id="heading-1-2"
      >
        <button
          aria-controls="collapse-1-2"
          aria-expanded="false"
          autocomplete="off"
          class="accordion-button collapsed"
          data-bs-target="#collapse-1-2"
          data-bs-toggle="collapse"
          type="button"
        >
          Accordion Item #2
        </button>
      </h2>
      <div
        aria-labelledby="heading-1-2"
        class="accordion-collapse collapse"
        data-bs-parent="#accordion-1"
        id="collapse-1-2"
        role="region"
      >
        <div
          class="accordion-body"
        >
          Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.
        </div>
      </div>
    </div>
    <div
      class="accordion-item"
    >
      <h2
        class="accordion-header"
        id="heading-1-3"
      >
        <button
          aria-controls="collapse-1-3"
          aria-expanded="false"
          autocomplete="off"
          class="accordion-button collapsed"
          data-bs-target="#collapse-1-3"
          data-bs-toggle="collapse"
          type="button"
        >
          Accordion Item #3
        </button>
      </h2>
      <div
        aria-labelledby="heading-1-3"
        class="accordion-collapse collapse"
        data-bs-parent="#accordion-1"
        id="collapse-1-3"
        role="region"
      >
        <div
          class="accordion-body"
        >
          Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.
        </div>
      </div>
    </div>
  </div>
</jest>
`;

exports[`OE - Accordion renders correctly 1`] = `
<jest>
  <div
    class="accordion"
    id="accordion-1"
  >
    <div
      class="accordion-item"
    >
      <h2
        class="accordion-header"
        id="heading-1-1"
      >
        <button
          aria-controls="collapse-1-1"
          aria-expanded="false"
          autocomplete="off"
          class="accordion-button"
          data-bs-target="#collapse-1-1"
          data-bs-toggle="collapse"
          type="button"
        >
          Accordion Item #1
        </button>
      </h2>
      <div
        aria-labelledby="heading-1-1"
        class="accordion-collapse collapse show"
        data-bs-parent="#accordion-1"
        id="collapse-1-1"
        role="region"
      >
        <div
          class="accordion-body"
        >
          Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.
        </div>
      </div>
    </div>
    <div
      class="accordion-item"
    >
      <h2
        class="accordion-header"
        id="heading-1-2"
      >
        <button
          aria-controls="collapse-1-2"
          aria-expanded="false"
          autocomplete="off"
          class="accordion-button collapsed"
          data-bs-target="#collapse-1-2"
          data-bs-toggle="collapse"
          type="button"
        >
          Accordion Item #2
        </button>
      </h2>
      <div
        aria-labelledby="heading-1-2"
        class="accordion-collapse collapse"
        data-bs-parent="#accordion-1"
        id="collapse-1-2"
        role="region"
      >
        <div
          class="accordion-body"
        >
          Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.
        </div>
      </div>
    </div>
    <div
      class="accordion-item"
    >
      <h2
        class="accordion-header"
        id="heading-1-3"
      >
        <button
          aria-controls="collapse-1-3"
          aria-expanded="false"
          autocomplete="off"
          class="accordion-button collapsed"
          data-bs-target="#collapse-1-3"
          data-bs-toggle="collapse"
          type="button"
        >
          Accordion Item #3
        </button>
      </h2>
      <div
        aria-labelledby="heading-1-3"
        class="accordion-collapse collapse"
        data-bs-parent="#accordion-1"
        id="collapse-1-3"
        role="region"
      >
        <div
          class="accordion-body"
        >
          Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.
        </div>
      </div>
    </div>
  </div>
</jest>
`;

exports[`OE - Accordion with custom title tags renders correctly 1`] = `
<jest>
  <div
    class="accordion"
    id="accordion-1"
  >
    <div
      class="accordion-item"
    >
      <h5
        class="accordion-header"
        id="heading-1-1"
      >
        <button
          aria-controls="collapse-1-1"
          aria-expanded="false"
          autocomplete="off"
          class="accordion-button"
          data-bs-target="#collapse-1-1"
          data-bs-toggle="collapse"
          type="button"
        >
          Accordion Item #1
        </button>
      </h5>
      <div
        aria-labelledby="heading-1-1"
        class="accordion-collapse collapse show"
        data-bs-parent="#accordion-1"
        id="collapse-1-1"
        role="region"
      >
        <div
          class="accordion-body"
        >
          Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.
        </div>
      </div>
    </div>
    <div
      class="accordion-item"
    >
      <h4
        class="accordion-header"
        id="heading-1-2"
      >
        <button
          aria-controls="collapse-1-2"
          aria-expanded="false"
          autocomplete="off"
          class="accordion-button collapsed"
          data-bs-target="#collapse-1-2"
          data-bs-toggle="collapse"
          type="button"
        >
          Accordion Item #2
        </button>
      </h4>
      <div
        aria-labelledby="heading-1-2"
        class="accordion-collapse collapse"
        data-bs-parent="#accordion-1"
        id="collapse-1-2"
        role="region"
      >
        <div
          class="accordion-body"
        >
          Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.
        </div>
      </div>
    </div>
    <div
      class="accordion-item"
    >
      <h3
        class="accordion-header"
        id="heading-1-3"
      >
        <button
          aria-controls="collapse-1-3"
          aria-expanded="false"
          autocomplete="off"
          class="accordion-button collapsed"
          data-bs-target="#collapse-1-3"
          data-bs-toggle="collapse"
          type="button"
        >
          Accordion Item #3
        </button>
      </h3>
      <div
        aria-labelledby="heading-1-3"
        class="accordion-collapse collapse"
        data-bs-parent="#accordion-1"
        id="collapse-1-3"
        role="region"
      >
        <div
          class="accordion-body"
        >
          Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.
        </div>
      </div>
    </div>
  </div>
</jest>
`;
